tests/ecmp/test_inner_hashing.py

```diff
@@ -147,41 +188,169 @@ def symmetric_hashing(duthosts, rand_one_dut_hostname):
+    config_vxlan_rule(duthost, " --ip-protocol {}", V4_ETHER_TYPE, "ipv4", inner_ipver)
+    config_nvgre_rule(duthost, " --ip-protocol {}", V4_ETHER_TYPE, "ipv4", inner_ipver)
+
+
+def config_ipv6_rules(duthost, inner_ipver):
+    config_vxlan_rule(duthost, " --ipv6-next-header {}", V6_ETHER_TYPE, "ipv6", inner_ipver)
+    config_nvgre_rule(duthost, " --ipv6-next-header {}", V6_ETHER_TYPE, "ipv6", inner_ipver)
+
+
+def config_vxlan_rule(duthost, ip_ver_option, ether_type, outer_ipver, inner_ipver):
+    inner_ether_type = V4_ETHER_TYPE if inner_ipver == "ipv4" else V6_ETHER_TYPE
+    duthost.command((ADD_PBH_RULE_BASE_CMD + ip_ver_option + VXLAN_L4_DST_PORT_OPTION)
+                    .format(TABLE_NAME,
+                            VXLAN_RULE_NAME.format(outer_ipver, inner_ipver),
+                            VXLAN_RULE_PRIO,
+                            ether_type,
+                            inner_ether_type,
+                            HASH_NAME,
+                            ECMP_PACKET_ACTION,
+                            VXLAN_IP_PROTOCOL,
+                            VXLAN_L4_DST_PORT))
+
+
+def config_nvgre_rule(duthost, ip_ver_option, ether_type, outer_ipver, inner_ipver):
+    inner_ether_type = V4_ETHER_TYPE if inner_ipver == "ipv4" else V6_ETHER_TYPE
+    duthost.command((ADD_PBH_RULE_BASE_CMD + ip_ver_option)
+                    .format(TABLE_NAME,
+                            NVGRE_RULE_NAME.format(outer_ipver, inner_ipver),
+                            NVGRE_RULE_PRIO,
+                            ether_type,
+                            inner_ether_type,
+                            HASH_NAME,
+                            ECMP_PACKET_ACTION,
+                            NVGRE_IP_PROTOCOL))
+
+
+def delete_ipv4_rules(duthost, inner_ipver):
+    delete_vxlan_nvgre_rules(duthost, "ipv4", inner_ipver)
+
+
+def delete_ipv6_rules(duthost, inner_ipver):
+    delete_vxlan_nvgre_rules(duthost, "ipv6", inner_ipver)
+
+
+def delete_vxlan_nvgre_rules(duthost, outer_ipver, inner_ipver):
+    duthost.command(DEL_PBH_RULE_CMD.format(TABLE_NAME, VXLAN_RULE_NAME.format(outer_ipver, inner_ipver)))
+    duthost.command(DEL_PBH_RULE_CMD.format(TABLE_NAME, NVGRE_RULE_NAME.format(outer_ipver, inner_ipver)))
+
+
+def test_inner_hashing(duthost, hash_keys, ptfhost, outer_ipver, inner_ipver, router_mac,
+                       vlan_ptf_ports, symmetric_hashing, build_fib, setup, dynamic_pbh):
     timestamp = datetime.now().strftime('%Y-%m-%d-%H:%M:%S')
-    log_file = "/tmp/inner_hash_test.InnerHashTest.{}.{}.log".format(ipver, timestamp)
+    log_file = "/tmp/inner_hash_test.InnerHashTest.{}.{}.{}.log".format(outer_ipver, inner_ipver, timestamp)
     logging.info("PTF log file: %s" % log_file)
 
-    inner_src_ip_range = SRC_IP_RANGE
-    inner_dst_ip_range = DST_IP_RANGE
-
-    if ipver == "ipv4":
+    if outer_ipver == "ipv4":
         outer_src_ip_range = SRC_IP_RANGE
         outer_dst_ip_range = DST_IP_RANGE
```

Comment:
why do we need to maintain two variables if they're always identical?

Reply:
because here 4 tests and parameters can be different:

outer - inner
ipv4 - ipv4
ipv4 - ipv6
ipv6 - ipv4
ipv6 - ipv6

Reply:
created function get_src_dst_ip_range